NuKeeper.Gitea/GiteaRestClient.cs

```diff
@@ -0,0 +1,189 @@
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web;
+using Newtonsoft.Json;
+using NuKeeper.Abstractions;
+using NuKeeper.Abstractions.Logging;
+using NuKeeper.Gitea.Model;
+
+namespace NuKeeper.Gitea
+{
+    public class GiteaRestClient
+    {
+        private readonly HttpClient _client;
+        private readonly INuKeeperLogger _logger;
+
+        public GiteaRestClient(HttpClient client, string token, INuKeeperLogger logger)
+        {
+            _client = client;
+            _logger = logger;
+
+            _client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            _client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("token", token);
+        }
+
+        /// <summary>
+        /// GET /user https://try.gitea.io/api/swagger#/user/userGetCurrent
+        /// </summary>
+        /// <returns>returns the current user</returns>
+        public async Task<User> GetCurrentUser()
+        {
+            return await GetResource<User>("user");
```

Comment:
This is correct? Just put in "user" for a request gets you the current user?

Reply:
yes https://try.gitea.io/api/swagger#/user/userGetCurrent
quite easy isn't it ;)
 GET /user